### PR TITLE
Airy: replace minor functions with direct calls

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

File: internal/amos/amos.go [hunk 1]
```diff
@@ -29,14 +29,14 @@ func iabs(a int) int {
 	return -a
 }
 
-func min0(a, b int) int {
+func min(a, b int) int {
```
Comment (sbinet) [inline internal/amos/amos.go:32]: for consistency with `iabs` above, call it `imin` instead? (and `imax` below?)

File: internal/amos/amos.go [hunk 2]
```diff
@@ -812,23 +814,23 @@ OneTen:
 		goto TwoNinety
 	}
 
-	STR = dexp(-ZR) * CSSR[KFLAG]
-	STI = -STR * dsin(ZI)
-	STR = STR * dcos(ZI)
-	tmp = zmlt(complex(COEFR, COEFI), complex(STR, STI))
+	STR = math.Exp(-ZR) * CSSR[KFLAG]
+	STI = -STR * math.Sin(ZI)
```
Comment (sbinet) [inline internal/amos/amos.go:820]: candidate for using `math.Sincos` ?

File: internal/amos/amos.go [hunk 3]
```diff
@@ -1109,20 +1111,20 @@ TwoSixtyOne:
...
+		tmp = cmplx.Log(complex(S2R, S2I))
 		STR = real(tmp)
 		STI = imag(tmp)
 		P2R = -ZDR + STR
 		P2I = -ZDI + STI
-		P2M = dexp(P2R) / TOL
-		P1R = P2M * dcos(P2I)
-		P1I = P2M * dsin(P2I)
+		P2M = math.Exp(P2R) / TOL
+		P1R = P2M * math.Cos(P2I)
```
Comment (sbinet) [inline internal/amos/amos.go:1129]: `math.Sincos` ?

File: internal/amos/amos.go [hunk 4]
```diff
@@ -1242,29 +1244,29 @@ func Zkscl(ZRR, ZRI, FNU float64, N int, YR, YI []float64, NZ int, RZR, RZI, ASC
...
 		CSR = real(tmp)
 		CSI = imag(tmp)
 		CSR = CSR - ZRR
 		CSI = CSI - ZRI
-		STR = dexp(CSR) / TOL
-		CSR = STR * dcos(CSI)
-		CSI = STR * dsin(CSI)
+		STR = math.Exp(CSR) / TOL
+		CSR = STR * math.Cos(CSI)
+		CSI = STR * math.Sin(CSI)
```
Comment (sbinet) [inline internal/amos/amos.go:1275]: `math.Sincos` ?

File: internal/amos/amos.go [hunk 5]
```diff
@@ -1315,23 +1317,23 @@ Twenty:
...
 		CSR = real(tmp)
 		CSI = imag(tmp)
 		CSR = CSR - ZDR
 		CSI = CSI - ZDI
-		STR = dexp(CSR) / TOL
-		CSR = STR * dcos(CSI)
-		CSI = STR * dsin(CSI)
+		STR = math.Exp(CSR) / TOL
+		CSR = STR * math.Cos(CSI)
+		CSI = STR * math.Sin(CSI)
```
Comment (sbinet) [inline internal/amos/amos.go]: `math.Sincos` ?

File: internal/amos/amos.go [hunk 6]
```diff
@@ -1459,15 +1461,15 @@ Fourty:
 		goto Fifty
 	}
 	YY = -ZNI
-	CSGNR = -CSGNI * dsin(YY)
-	CSGNI = CSGNI * dcos(YY)
+	CSGNR = -CSGNI * math.Sin(YY)
+	CSGNI = CSGNI * math.Cos(YY)
```
Comment (sbinet) [inline internal/amos/amos.go:1475]: `math.Sincos` ?

File: internal/amos/amos.go [hunk 7]
```diff
@@ -1459,15 +1461,15 @@ Fourty:
...
 Fifty:
 	// CALCULATE CSPN=EXP(FNU*PI*I) TO MINIMIZE LOSSES OF SIGNIFICANCE
 	// WHEN FNU IS LARGE
 	INU = int(float32(FNU))
 	ARG = (FNU - float64(float32(INU))) * SGN
-	CSPNR = dcos(ARG)
-	CSPNI = dsin(ARG)
+	CSPNR = math.Cos(ARG)
+	CSPNI = math.Sin(ARG)
```
Comment (sbinet) [inline internal/amos/amos.go:1483]: `math.Sincos` ?

File: internal/amos/amos.go [hunk 8]
```diff
@@ -1583,8 +1585,8 @@ Twenty:
 	INU = int(float32(FNU))
 	ARG = (FNU - float64(float32(INU))) * PI
 	INU = INU + N - IL
-	AK = -dsin(ARG)
-	BK = dcos(ARG)
+	AK = -math.Sin(ARG)
+	BK = math.Cos(ARG)
```
Comment (sbinet) [inline internal/amos/amos.go:1602]: `math.Sincos` ?

File: internal/amos/amos.go [hunk 9]
```diff
@@ -2002,14 +2004,14 @@ Fourty:
...
 Fifty:
-	AA = dexp(AK1R)
+	AA = math.Exp(AK1R)
 	if IFLAG == 1 {
 		AA = AA * SS
 	}
-	COEFR = AA * dcos(AK1I)
-	COEFI = AA * dsin(AK1I)
+	COEFR = AA * math.Cos(AK1I)
+	COEFI = AA * math.Sin(AK1I)
```
Comment (sbinet) [inline internal/amos/amos.go]: `math.Sincos` ?

**Comment by sbinet** [comment]
In a previous HEP experiment, we had good results (performance-wise) migrating a bunch of `sin`+`cos` calls to `sincos` (the `gcc` version we were using had special assembly support, and later on, the `64b` version had also good native support).
I haven't checked whether using `math.Sincos` in `Go` was a win or not...

LGTM otherwise. (feel free to ignore my comment wrt `min`/`imin` and `max`/`imax`)

File: internal/amos/amos.go (hunk 1 above)
Comment (btracey) [inline internal/amos/amos.go:32]: Changed `iabs` to `abs` instead

**Comment by btracey** [comment]
The numeric noise in this code are pretty bad. I replaced the Sincoses that don't cause huge deviations in the final result. The other ones I left commented for future fixes, once we feel we no longer need comparison with the old code to check changes.

**Pull request merged**